packages/flame/lib/src/widgets/sprite_widget.dart

```diff
@@ -29,20 +31,25 @@ class SpriteWidget extends StatelessWidget {
   /// A builder function that is called while the loading is on the way
   final WidgetBuilder? loadingBuilder;
 
-  final Future<Sprite> Function() _spriteFuture;
+  final FutureOr<Sprite> _spriteFuture;
 
-  SpriteWidget({
+  const SpriteWidget({
     required Sprite sprite,
     this.anchor = Anchor.topLeft,
     this.angle = 0,
     this.srcPosition,
     this.srcSize,
-    this.errorBuilder,
-    this.loadingBuilder,
     Key? key,
-  })  : _spriteFuture = (() => Future.value(sprite)),
+  })  : _spriteFuture = sprite,
+        errorBuilder = null,
+        loadingBuilder = null,
         super(key: key);
 
+  /// loads image from asset [path] and renders it as Widget.
+  /// it has [loadingBuilder] while the image from [path] loads image
+  /// To render without loading or gapless playback when the [path] change,
+  /// consider load an image from somewhere and directly pass it to
+  /// default constructor
```

Comment:
```suggestion
  /// Load the image from the asset [path] and renders it as a widget.
  ///
  /// It will use the [loadingBuilder] while the image from [path] is loading.
  /// To render without loading, or when you want to have a gapless playback 
  /// when the [path] value changes, consider loading the image beforehand 
  /// and direct pass it to the default constructor.
```